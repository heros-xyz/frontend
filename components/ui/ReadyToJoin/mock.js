export const mock = {
  content: {
    title: "Are you ready to join?",
    description:
      "Donâ€™t miss this amazing opportunity to shorten the gap between athletes and fans.",
    banner: {
      fanTitle: "become a fan",
      fanBanner: "/images/athlete-banner.png",
      athleteBanner: "/images/fan-banner.png",
      athleteTitle: "become an athlete"
    }
  }
};