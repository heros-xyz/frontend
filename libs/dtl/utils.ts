import { useState, useEffect, useCallback, useRef, useMemo } from "react";
import { DocumentData, Query, startAfter, query, getDocs } from "firebase/firestore";

// Test this custom hook, when implement infinite scroll
export const usePaginateQuery = (queryFn: () => Query) => {
    const [data, setData] = useState<null | DocumentData[]>(null)
    const isMountedRef = useRef(false);
    const lastItemRef = useRef<any>(null);
    const [isLoading, setIsLoading] = useState<boolean>();
    const [errorMsg, setErrorMsg] = useState();

    const resetStates = () => {
        setData(null);
        setData([]);
        setIsLoading(false);
    };

    useEffect(() => {
        if (isMountedRef.current === true) return;

        async function fetchQuery() {
            try {
                isMountedRef.current = true;
                setIsLoading(true);
                const q = query(queryFn());
                const querySnapshot = await getDocs(q);
                setData([...querySnapshot.docs]);
                if (lastItemRef) {
                    lastItemRef.current = querySnapshot.docs[querySnapshot.docs.length - 1];
                }
                setIsLoading(false);
            } catch (error) {
                resetStates();
                setErrorMsg(error.code);
            }
        }
        fetchQuery();
    }, [queryFn]);

    const more = useCallback(async () => {
        try {
            setIsLoading(true);
            const next = query(queryFn(), startAfter(lastItemRef.current));
            const querySnapshot = await getDocs(next);
            setData([...data ?? [], ...querySnapshot.docs]);
            lastItemRef.current = querySnapshot.docs[querySnapshot.docs.length - 1];
            setIsLoading(false);
        } catch (error) {
            resetStates();
            setErrorMsg(error.code);
        }
    }, [data, queryFn]);

    return useMemo(
        () => ({
            more,
            isLoading,
            data,
            errorMsg,
        }),
        [more, isLoading, data, errorMsg]
    );
};