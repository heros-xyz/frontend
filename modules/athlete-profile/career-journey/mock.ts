export const CAREER_JOURNEY_DATA = [
  {
    title: "vvvv",
    description: "vvvv",
    from: "2022-11-29",
    to: null,
    isArchive: true,
    isCurrent: true,
    icon: "cricket",
  },
  {
    title: "Ggggh",
    description: "",
    from: "2012-01-24",
    to: "2022-02-20",
    isArchive: true,
    isCurrent: false,
    icon: "flag",
  },
  {
    title: "Hhh",
    description: "Uu",
    from: "1991-03-30",
    to: null,
    isArchive: true,
    isCurrent: false,
    icon: "golf",
  },
  {
    title: "Hh",
    description: "Hhhh",
    from: "2021-03-29",
    to: "2021-03-30",
    isArchive: true,
    isCurrent: false,
    icon: "soccer",
  },
  {
    title: "Ghjkkj ghh",
    description: "Ghi kkk",
    from: "1997-10-29",
    to: "2021-10-29",
    isArchive: true,
    isCurrent: false,
    icon: "soccer",
  },
  {
    title: "Phong",
    description: "",
    from: "1980-02-29",
    to: null,
    isArchive: true,
    isCurrent: false,
    icon: "golf",
  },
  {
    title: "My nam is Phong",
    description:
      "Hhh ahhâj dkf nffn fkfk hâhh dpgong jfjffn bxbxxb abb có gò nưav đâu mà ",
    from: "2022-10-30",
    to: "2023-10-30",
    isArchive: true,
    isCurrent: false,
    icon: "medal",
  },
  {
    title: "Hhh",
    description: "Hhh",
    from: "2021-10-30",
    to: null,
    isArchive: true,
    isCurrent: false,
    icon: "soccer",
  },
];
