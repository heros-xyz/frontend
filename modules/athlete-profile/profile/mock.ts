export const basicInfo = {
  id: "c2b7f17a-ee39-4b69-a35f-4ec8cf6c7dcb",
  createdAt: "2023-01-17T05:46:06.812Z",
  updatedAt: "2023-02-03T11:26:19.748Z",
  deletedAt: null,
  nickName: "dds",
  story:
    "abc def ghi jkl mno pqrs tuv wxyz ABC DEF GHI JKL NO PQRS TUV WXYZ !\"§ $%& /() =?* '<> #|; ²³~ @`´ ©«» ¤¼× {} abc def ghi jkl mno pqrs tuv wxyz ABC DEF GHI JKL MNO PQRS TUV WXYZ !\"§ $%& /() =?* '<> #|; ²³~ @`´ ©«» ¤¼× {} abc def ghi jkl mno pqrs tuv wxyz ABC DEF GHI JKL MNO PQRS TUV WXYZ !\"§ $%& /() =?* '<> #|; ²³~ @`´ ©«» ¤¼× {} abc def ghi jkl mno pqrs tuv wxyz ABC DEF GHI JKL MNO PQRS TUV WXYZ !\"§ $%& /() =?* '<> #|; ²³~ @`´ ©«» ¤¼× {} abc def ghi jkl mno pqrs tuv wxyz ABC DEF GHI JKL MNO PQR",
  nationality: {
    id: "58cd8f52-76b0-46dd-9a29-8cef2dbcfc74",
    createdAt: "2023-01-16T10:22:40.062Z",
    updatedAt: "2023-01-16T10:22:40.062Z",
    deletedAt: null,
    name: "Albanian",
    code: "8",
    phoneCode: "355",
  },
  dateOfBirth: "2022-11-29T00:00:00.000Z",
  gender: 0,
};

export const sportProfile = {
  data: {
    id: "dc3fb3e3-b659-427f-875b-4fe5549c71fb",
    createdAt: "2023-01-17T08:49:40.080Z",
    updatedAt: "2023-01-17T08:49:40.080Z",
    sportProfilesItems: [],
    goal: "fff",
    currentTeam: "dsadasa",
  },
  success: true,
};

export const basicTest = {
  id: "c2b7f17a-ee39-4b69-a35f-4ec8cf6c7dcb",
  createdAt: "2023-01-17T05:46:06.812Z",
  updatedAt: "2023-02-03T11:26:19.748Z",
  deletedAt: null,
  nickName: "dds",
  story:
    "abc def ghi jkl mno pqrs tuv wxyz ABC DEF GHI JKL NO PQRS TUV WXYZ !\"§ $%& /() =?* '<> #|; ²³~ @`´ ©«» ¤¼× {} abc def ghi jkl mno pqrs tuv wxyz ABC DEF GHI JKL MNO PQRS TUV WXYZ !\"§ $%& /() =?* '<> #|; ²³~ @`´ ©«» ¤¼× {} abc def ghi jkl mno pqrs tuv wxyz ABC DEF GHI JKL MNO PQRS TUV WXYZ !\"§ $%& /() =?* '<> #|; ²³~ @`´ ©«» ¤¼× {} abc def ghi jkl mno pqrs tuv wxyz ABC DEF GHI JKL MNO PQRS TUV WXYZ !\"§ $%& /() =?* '<> #|; ²³~ @`´ ©«» ¤¼× {} abc def ghi jkl mno pqrs tuv wxyz ABC DEF GHI JKL MNO PQR",
  nationality: {
    id: "58cd8f52-76b0-46dd-9a29-8cef2dbcfc74",
    createdAt: "2023-01-16T10:22:40.062Z",
    updatedAt: "2023-01-16T10:22:40.062Z",
    deletedAt: null,
    name: "Albanian",
    code: "8",
    phoneCode: "355",
  },
  dateOfBirth: "2022-11-29T00:00:00.000Z",
  gender: 0,
  firstName: "a",
  lastName: "b",
};
