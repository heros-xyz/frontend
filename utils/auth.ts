import { getCookie, setCookie } from "cookies-next";
import { NextApiRequest, NextApiResponse } from "next";
import { signOut } from "next-auth/react";
import { IAuthResponse, IToken } from "@/types/users/types";
import { $http } from "@/libs/http";
import { store } from "@/store";
import { finishLoading, startLoading } from "@/store/globalSlice";

export const updateSession = async () => {
  await fetch("/api/auth/session?update");
  const event = new Event("visibilitychange");
  document.dispatchEvent(event);
};

export const getAccessTokenFromCookie = (
  req: NextApiRequest,
  res: NextApiResponse
) => {
  const accessToken = getCookie("_Auth.access-token", {
    req,
    res,
    path: "/",
  });

  return accessToken;
};

export const getUserRoleFromCookie = (
  req: NextApiRequest,
  res: NextApiResponse
) => {
  const userRole = getCookie("role", {
    req,
    res,
    path: "/",
  });

  return userRole;
};

export const setAccessTokenToCookie = (
  credentials: IToken,
  req: NextApiRequest,
  res: NextApiResponse
) => {
  const { accessToken, refreshToken, refreshExpiresIn } = credentials;
  const expire = new Date().getTime() + 1000 * 24 * 60 * 60;
  const refreshExpire = new Date().getTime() + +refreshExpiresIn * 1000;

  setCookie("_Auth.access-token", accessToken, {
    req,
    res,
    path: "/",
    // httpOnly: true,
    // secure: process.env.NODE_ENV !== 'development',
    expires: new Date(expire),
  });

  setCookie("_Auth.refresh-token", refreshToken, {
    req,
    res,
    path: "/",
    expires: new Date(refreshExpire),
  });
};

export const fetchUser = async (accessToken: string) => {
  try {
    const res = await fetch(`${process.env.HEROS_BASE_URL}/auth/me`, {
      method: "GET",
      credentials: "include",
      headers: {
        Authorization: `Bearer ${accessToken}`,
        "X-FP-API-KEY": "iphone",
        "Content-Type": "application/json",
      },
    });

    const user = await res.json();

    return Promise.resolve(user);
  } catch (error) {
    return Promise.reject(error);
  }
};

export const signInSocial = async (
  data: {
    token: string | undefined;
    role: "FAN" | "ATHLETE";
    provider: string;
  },
  req: NextApiRequest,
  res: NextApiResponse
) => {
  try {
    const url =
      data.provider === "google"
        ? "/auth/google-authenticate"
        : "/auth/facebook-authenticate";
    const loginResp = await $http.post<unknown, IAuthResponse>(
      `${process.env.HEROS_BASE_URL}${url}`,
      data
    );

    if (loginResp) {
      const { accessToken } = loginResp.token;
      setAccessTokenToCookie(loginResp.token, req, res);

      const user = await fetchUser(accessToken);
      return Promise.resolve(user);
    }
  } catch (error) {
    return Promise.reject(error);
  }
};

export const onSignOut = async () => {
  store.dispatch(startLoading());
  await Promise.all([
    signOut({
      redirect: true,
      callbackUrl: "/",
    }),
    $http({
      baseURL: "",
      url: `/api/remove-authorization`,
    }),
    $http({
      method: "POST",
      baseURL: "",
      url: `/api/auth/sign-out`,
    }),
    ,
  ]);

  if (typeof window !== undefined) {
    setTimeout(() => {
      store.dispatch(finishLoading());
    }, 200);
  }
};
