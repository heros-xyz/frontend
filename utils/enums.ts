export enum HttpErrorCode {
  // Auth sections -> start from 1000
  USER_NOT_REGISTERED = 1000,
  USER_ALREADY_REGISTERED = 1001,
  EXPIRED_OTP = 1002,
  INVALID_OTP = 1003,
  GOOGLE_TOKEN_INVALID = 1004,
  FACEBOOK_TOKEN_INVALID = 1005,
  TOKEN_INVALID = 1006,
  TOKEN_REVOKED = 1007,
  USER_ALREADY_REGISTERED_OTHER_ROLE = 1008,
  MUST_SIGN_UP_FIRST = 1009,

  // Interaction section -> start from 2000
  INTERACTION_NOT_FOUND = 2000,
}

export enum NotificationEventType {
  FAN_SUBSCRIBE_ATHLETE = "F_SUBSCRIBE", // A fan subscribes to an athlete's tier // Good
  FAN_LIKE_INTERACTION = "F_LIKE_INTERACTION", // A fan likes an athlete's interactions // Good
  FAN_COMMENT_INTERACTION = "F_COMMENT_INTERACTION", // A fan comments on an athlete's interaction // Good
  FAN_LIKE_COMMENT = "F_LIKE_COMMENT", // A fan likes an athlete's comment & A fan likes a comment's reply. // Good
  FAN_REPLY_COMMENT = "F_REPLY_COMMENT", // A fan replies to an athlete's comment // Good
  FAN_LIKE_REPLY = "F_LIKE_REPLY", // A fan replies to an athlete's comment // Good

  ATHLETE_NEW_INTERACTION = "A_NEW_INTERACTION", // An athlete posts a new interaction // Good
  ATHLETE_LIKE_INTERACTION = "A_LIKE_INTERACTION",
  ATHLETE_COMMENT_INTERACTION = "A_COMMENT_INTERACTION", // An athlete comments an interaction
  ATHLETE_LIKE_COMMENT = "A_LIKE_COMMENT", // An athlete likes fan's comment // Good
  ATHLETE_REPLY_COMMENT = "A_REPLY_COMMENT", // An athlete replies to fan's comment // Good
  ATHLETE_LIKE_REPLY = "A_LIKE_REPLY", // An athlete likes fan's comment's reply // Good

  FAN_LIKE_FAN_COMMENT = "F_LIKE_F_COMMENT", // Another fan likes fan's comment // Good
  FAN_REPLY_FAN_COMMENT = "F_REPLY_F_COMMENT", // Another fan replies to fan's comment // Good
  FAN_LIKE_FAN_REPLY = "F_LIKE_F_REPLY", // Another fan likes fan's comment's reply // Good
}

export enum CorporateWebsiteLink {
  TERM_AND_CONDITION = "term-of-service",
  PRIVACY_POLICY = "privacy-policy",
}
