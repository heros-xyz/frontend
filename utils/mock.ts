export const notiList = [
  {
    id: "1",
    type: "A_NEW_INTERACTION",
    message: "notification.athlete-new-interaction",
    source: {
      id: "48",
      avatar: "https://cdn-icons-png.flaticon.com/512/3135/3135715.png",
      fullName: "Sam Smith",
    },
    interaction: {
      id: "70",
      content: "some content should be trimmed if too long",
    },
    comment: null,
    readAt: null,
    createdAt: "2023-02-22T08:43:57.555Z",
  },
  {
    id: "3",
    type: "A_NEW_INTERACTION",
    message: "notification.athlete-new-interaction",
    source: {
      id: "4",
      avatar: "https://cdn-icons-png.flaticon.com/512/3135/3135715.png",
      fullName: "Sam Smith",
    },
    interaction: {
      id: "35",
      content: "some content should be trimmed if too long",
    },
    comment: null,
    readAt: null,
    createdAt: "2023-02-22T08:43:57.555Z",
  },
];

export const fanNotiList = [
  {
    id: "1",
    type: "A_NEW_INTERACTION",
    message: "notification.athlete-new-interaction",
    source: {
      id: "48",
      avatar: "https://cdn-icons-png.flaticon.com/512/3135/3135715.png",
      fullName: "Sam Smith",
    },
    interaction: {
      id: "70",
      content: "some content should be trimmed if too long",
    },
    comment: null,
    readAt: null,
    createdAt: "2023-02-22T08:43:57.555Z",
  },
  {
    id: "2",
    type: "A_NEW_INTERACTION",
    message: "notification.athlete-new-interaction",
    source: {
      id: "4",
      avatar: "https://cdn-icons-png.flaticon.com/512/3135/3135715.png",
      fullName: "Sam Smith",
    },
    interaction: {
      id: "35",
      content: "some content should be trimmed if too long",
    },
    comment: null,
    readAt: null,
    createdAt: "2023-02-22T08:43:57.555Z",
  },
];

export const sports = [
  {
    value: "Football",
    label: "Football",
  },
  {
    value: "Basketball",
    label: "Basketball",
  },
  {
    value: "Badminton",
    label: "Badminton",
  },
  {
    value: "Cricket",
    label: "Cricket",
  },
  {
    value: "Swimming",
    label: "Swimming",
  },
  {
    value: "Tennis",
    label: "Tennis",
  },
  {
    value: "Baseball",
    label: "Baseball",
  },
  {
    value: "Running",
    label: "Running",
  },
  {
    value: "Flexing",
    label: "Flexing",
  },
];

export const timelineItems = [
  {
    title: "Join “Jaworznickie Stowarzyszenie Piłkarskie Szczakowianka Club",
    description:
      "I became the best new player on team I became the best new player on team I became the best new player on team I became the best new player on team",
    startDate: "10/01/2022",
    endDate: "10/24/2022",
    isPeriodDate: true,
    icon: "FLAG",
  },
];
