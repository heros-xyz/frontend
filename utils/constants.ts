import { HttpErrorCode } from "./enums";

export const AUTH_ERROR_CODE = {
  [HttpErrorCode.MUST_SIGN_UP_FIRST]: `This email hasn't been registered. Please sign up.`,
  [HttpErrorCode.USER_ALREADY_REGISTERED]: `This email has been registered. Please sign in.`,
  [HttpErrorCode.INVALID_OTP]: "Invalid OTP, please recheck!",
  [HttpErrorCode.EXPIRED_OTP]: "Expired OTP, please recheck!",
};

export const getAuthErrorCode = (code: string | number | undefined) => {
  return AUTH_ERROR_CODE[code as keyof typeof AUTH_ERROR_CODE];
};

export const ACTIVE_PATHS = [
  "/athlete",
  "/fan",
  "/athlete/notification",
  "/fan/notification",
  "/athlete/interactions",
  "/fan/interactions",
  "/athlete/my-fan",
  "/athlete/my-profile",
  "/fan/my-profile",
];

export const gaMeasurementId = "G-W55H90J71S"

export const ADMIN_ROLE = 'ADMIN'
export const ATHLETE_ROLE = 'ATHLETE'
export const FAN_ROLE = 'FAN'
export const REFRESH_TOKEN_KEY = '_Auth.refresh-token'
export const ACCESS_TOKEN_KEY = '_Auth.access-token'
