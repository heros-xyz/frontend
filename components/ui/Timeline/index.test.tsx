import { render } from "@testing-library/react";
import { timelineItems } from "@/utils/mock";
import TimeLine from "./index";

it("renders homepage unchanged", () => {
  const { container } = render(
    <TimeLine isAddJourney bgColor={"primary"} items={timelineItems} />
  );
  expect(container).toMatchSnapshot();
});
