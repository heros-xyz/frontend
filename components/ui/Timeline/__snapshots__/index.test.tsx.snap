// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <div
    class="css-e25rbx"
  >
    <div
      class="css-19f3oqx"
    >
      <div
        class="css-70qvj9"
      >
        <div
          class="css-f1zsqn"
        >
          <div
            class="css-46mybu"
          >
            <svg
              class="chakra-icon css-xg0vmv"
              focusable="false"
              viewBox="0 0 12 14"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5.16699 1.99998H3.50033V0.333313H5.16699V1.99998ZM8.50033 0.333313H6.83366V1.99998H8.50033V0.333313ZM3.50033 8.66665H5.16699V6.99998H3.50033V8.66665ZM11.8337 5.33331V3.66665H10.167V5.33331H11.8337ZM11.8337 8.66665V6.99998H10.167V8.66665H11.8337ZM6.83366 8.66665H8.50033V6.99998H6.83366V8.66665ZM11.8337 0.333313H10.167V1.99998H11.8337V0.333313ZM6.83366 3.66665V1.99998H5.16699V3.66665H6.83366ZM1.83366 5.33331V3.66665H3.50033V1.99998H1.83366V0.333313H0.166992V13.6666H1.83366V6.99998H3.50033V5.33331H1.83366ZM8.50033 6.99998H10.167V5.33331H8.50033V6.99998ZM5.16699 5.33331V6.99998H6.83366V5.33331H5.16699ZM3.50033 3.66665V5.33331H5.16699V3.66665H3.50033ZM6.83366 5.33331H8.50033V3.66665H6.83366V5.33331ZM8.50033 1.99998V3.66665H10.167V1.99998H8.50033Z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
        <div
          class="css-bt25en"
        >
          <div
            class="css-1agcy0f"
          >
            <div
              class="css-mlzgdd"
            />
          </div>
          <div
            class="css-1u3qjd0"
          >
            <div
              class="css-7aj1ay"
            />
            <div
              class="css-52ky4n"
            >
              <button
                class="chakra-button css-b1wgcr"
                type="button"
              >
                <svg
                  class="chakra-icon css-17cusq9"
                  focusable="false"
                  viewBox="0 0 32 32"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M31 16C31 24.2843 24.2843 31 16 31C7.71573 31 1 24.2843 1 16C1 7.71573 7.71573 1 16 1C24.2843 1 31 7.71573 31 16Z"
                  />
                  <path
                    d="M16 11V16M16 16V21M16 16H21M16 16H11M31 16C31 24.2843 24.2843 31 16 31C7.71573 31 1 24.2843 1 16C1 7.71573 7.71573 1 16 1C24.2843 1 31 7.71573 31 16Z"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="1.5"
                  />
                </svg>
              </button>
              <p
                class="chakra-text css-3c892c"
              >
                Add as many as you like to tell your journey
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="css-q9og19"
      >
        <div
          class="css-1jaf54"
        >
          <h2
            class="chakra-heading css-109tgpy"
          >
            Join “Jaworznickie Stowarzyszenie Piłkarskie Szczakowianka Club
          </h2>
          <p
            class="chakra-text css-s2a0zi"
          >
            I became the best new player on team I became the best new player on team I became the best new player on team I became the best new player on team
          </p>
        </div>
        <div
          class="css-64rv73"
        >
          <div
            class="css-174s0jl"
          />
          <div
            class="css-1kmzqxq"
          >
            <div
              class="css-s2a0zi"
            >
              <span
                class="chakra-text css-0"
              >
                1
              </span>
              <sup
                class="chakra-text css-cky0vs"
              >
                st
              </sup>
              <span
                class="chakra-text css-0"
              >
                 Oct 2022
              </span>
            </div>
            <div
              class="css-1vp8jwk"
            >
              <div
                class="css-1nz5tvn"
              />
              <div
                class="css-4qxy49"
              >
                <svg
                  class="chakra-icon css-q21vef"
                  focusable="false"
                  viewBox="0 0 14 14"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5 7.5L7 9.5M7 9.5L9 7.5M7 9.5L7 4.5M1 7C1 3.68629 3.68629 0.999999 7 0.999999C10.3137 1 13 3.68629 13 7C13 10.3137 10.3137 13 7 13C3.68629 13 1 10.3137 1 7Z"
                    stroke="#313F4C"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="0.5"
                  />
                </svg>
              </div>
              <div
                class="css-159arjl"
              />
            </div>
            <div
              class="css-s2a0zi"
            >
              <span
                class="chakra-text css-0"
              >
                24
              </span>
              <sup
                class="chakra-text css-cky0vs"
              >
                th
              </sup>
              <span
                class="chakra-text css-0"
              >
                 Oct 2022
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
